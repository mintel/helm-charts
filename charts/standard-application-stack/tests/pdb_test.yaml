suite: Test Pod Disruption Budgets
templates:
  - pdbs.yaml
capabilities:
  majorVersion: 1
  minorVersion: 31
release:
  namespace: test-namespace
tests:
  - it: Creates a PDB when replicas > 1
    set:
      global.clusterEnv: qa
      replicas: 2
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: PodDisruptionBudget
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app

  - it: Does not create a PDB when replicas = 1
    set:
      global.clusterEnv: qa
      replicas: 1
    asserts:
      # - matchSnapshot: {} # Check for regressions and unexpected changes.
      - hasDocuments:
          count: 0

  - it: Creates a PDB when autoscaling is disabled and autoscaling.minReplicaCount is set
    set:
      global.clusterEnv: qa
      autoscaling.enabled: false
      autoscaling.minReplicaCount: 2
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: PodDisruptionBudget
      - hasDocuments:
          count: 1

  - it: Does not create a PDB when autoscaling.minReplicaCount  = 1 and autoscaling is enabled
    set:
      global.clusterEnv: qa
      replicas: 2
      autoscaling.enabled: true
      autoscaling.minReplicaCount: 1
    asserts:
      # - matchSnapshot: {} # Check for regressions and unexpected changes.
      - hasDocuments:
          count: 0
