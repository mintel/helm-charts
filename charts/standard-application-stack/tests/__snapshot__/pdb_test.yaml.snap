Creates a PDB when autoscaling is disabled and autoscaling.minReplicaCount is set:
  1: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      minAvailable: 50%
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: example-app
Creates a PDB when replicas > 1:
  1: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      minAvailable: 50%
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: example-app
